Update machine_detail_widget1.php

precursorca opened the pull request:
Rewrite code to get icons from https://statici.icloud.com/fmipmobile/deviceImages-9.0/ for all Intel and Apple Silicon Macs including the correct color obtained from the iBridge module. Exemptions are provided for a few computers that don't follow the rules properly including the iMac Pro 1,1 and the 2022 13" MacBook Air Starlight model.

precursorca committed:
Update machine_detail_widget1.php

Rewrite code to get icons from https://statici.icloud.com/fmipmobile/deviceImages-9.0/ for all Intel and Apple Silicon Macs including the correct color obtained from the iBridge module.
Exemptions are provided for a few computers that don't follow the rules properly including the iMac Pro 1,1 and the 2022 13" MacBook Air Starlight model.

tuxudo commented:
This is contains some of the same logic/functionality as the ibridge module. If the admin has the ibridge module installed, it will automatically overwrite/update the model image as of v3.1 (August 2023) of the ibridge module. This PR also breaks that functionality on line 4, this should be amended as to not break the ibridge module. 

We should also not assume the admin has the non-default ibridge module installed/enabled and instead rewrite the PR as if the ibridge module is unavailable, letting that module update the color if it is available.

precursorca commented:
I see what you mean. 
Really, it would be more elegant to update the model_lookup_helper.php.
But that seems beyond my skill at the moment.
As far as this code goes we could probably eliminate the search for a color altogether and just pull the generic icon.
That would mean no iBridge conflict I think.

precursorca committed:
Update machine_detail_widget1.php

Removed reference to the )optional) ibridge module. The icon's color is therefore no longer specified and the generic icon is pulled instead.

precursorca commented:
Removed all reference to the (optional) ibridge module. Color of icon is therefore no longer specified and a generic icon is pulled instead.
